lib/migrator.js

```diff
@@ -48,27 +68,32 @@ internals.executeMigration = async (migration, rollback) => {
     catch (err) {
 
         // Even if our symlinking fails, we need to manually cleanup our temporary directory because
-        // the unsafeCleanup: true option we pass to tempDirAsync (Tmp.dir) causes the tool to not delete the temp directory
-        // if it has entries, which it will if we fail partway through
-        tmpResults.cleanup();
+        // the unsafeCleanup: true option we pass to tempDir (Tmp.dir promisified) causes Tmp to not delete the temp directory if it has entries,
+        // which it will if we fail partway through
+        tmpCleanup();
+        throw err;
     }
 
-    // TODO Review / fix tmpAssets.tmpPath
-    const relMigrationsDir = Path.relative(process.cwd(), tmpResults.tmpPath);
+    const relMigrationsDir = Path.relative(process.cwd(), tmpPath);
     const latestOrRollback = rollback ? 'rollback' : 'latest';
 
     await knex.migrate[latestOrRollback]({ directory: relMigrationsDir });
```

Comment:
If this line throws, will `tmpCleanup()` be called?  You might want a `finally {}` block for pretty much this whole function.